db/db_badger.go

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) 2018 IoTeX
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package db
+
+import (
+	"context"
+	"sync"
+
+	"github.com/dgraph-io/badger"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/config"
+)
+
+// badgerDB is KVStore implementation based bolt DB
+type badgerDB struct {
+	mutex  sync.RWMutex
```

Comment:
I don't think the mutex is necessary. DB usually has its own concurrency control

Reply:
i also think not necessary, will confirm and remove in another PR